pkg/governance-scripts/contracts/20230109-gauge-adder-migration-v2-to-v3/GaugeAdderMigrationCoordinator.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IArbitrumFeeProvider.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IOptimismGasLimitProvider.sol";
+
+import "../BaseCoordinator.sol";
+
+contract GaugeAdderMigrationCoordinator is BaseCoordinator {
+    IGaugeAdder public immutable newGaugeAdder;
+    IGaugeAdder public immutable oldGaugeAdder;
+
+    IGaugeController public immutable gaugeController;
+
+    ILiquidityGaugeFactory public immutable ethereumGaugeFactory;
+    ILiquidityGaugeFactory public immutable polygonRootGaugeFactory;
+    ILiquidityGaugeFactory public immutable arbitrumRootGaugeFactory;
+    ILiquidityGaugeFactory public immutable optimismRootGaugeFactory;
+
+    address public immutable liquidityMiningCommitteeMultisig;
+    address public immutable gaugeCheckpointingMultisig;
+
+    constructor(
+        IAuthorizerAdaptor authorizerAdaptor,
```

Comment:
soft comment: we could also use the adaptor entrypoint here. The result is the same since the adaptor entrypoint forwards the call to the adaptor, but it'd be a reminder that there's no reason to interact with the adaptor directly. 

Feel free to skip the comment if you think it's redundant.

Reply:
This requires the authorizer adaptor endpoint to be authorized to make the relevant call on the authorizer adaptor as well as the migrator being authorized to call the authorizer adaptor endpoint. It's simpler to just call the adaptor directly from contracts for the time being.